ape_safe/client.py

```diff
@@ -165,72 +271,158 @@ def _all_transactions(self) -> Iterator[dict]:
-        if sigs:
-            tx_data.signatures = HexBytes(
-                reduce(lambda raw_sig, next_sig: raw_sig + next_sig.encode_vrs(), sigs, b"")
-            )
+        if not response.ok:
+            raise ClientResponseError(url, response)
 
-        url = f"{self.transaction_service_url}/api/v1/multisig-transactions"
-        response = requests.post(url, json=tx_data.dict())
+    def post_signature(
+        self,
+        safe_tx_or_hash: Union[SafeTx, SafeTxID],
+        signer: AddressType,
+        signature: MessageSignature,
+    ):
+        if isinstance(safe_tx_or_hash, SafeTx):
+            safe_tx = safe_tx_or_hash
+            safe_tx_hash = hash_eip712_message(safe_tx).hex()
+        else:
+            safe_tx_hash = safe_tx_or_hash
 
-        if not response.ok:
-            raise
+        if not isinstance(safe_tx_hash, str):
+            raise TypeError("Expecting str-like type for 'safe_tx_hash'.")
 
-    def post_signature(self, safe_tx_id: SafeTxID, signature: MessageSignature):
         url = (
-            f"{self.transaction_service_url}"
-            f"/api/v1/multisig-transactions/{safe_tx_id.hex()}/confirmations"
+            f"{self.transaction_service_url}/api"
+            f"/v1/multisig-transactions/{safe_tx_hash}/confirmations"
+        )
+        response = requests.post(
+            url, json={"origin": "ApeWorX/ape-safe", "signature": signature.encode_rsv().hex()}
         )
-        response = requests.post(url, json={"signature": signature.encode_vrs().hex()})
 
         if not response.ok:
-            raise
+            if "The requested resource was not found on this server" in response.text:
+                raise MultisigTransactionNotFoundError(safe_tx_hash, url, response)
+
+            raise ClientResponseError(url, response)
+
+
+class MockSafeClient(BaseSafeClient, ManagerAccessMixin):
+    def __init__(self, contract: ContractInstance):
+        self.contract = contract
+        self.transactions: Dict[Union[SafeTx, SafeTxID], SafeApiTxData] = {}
+        self.transactions_by_nonce: Dict[int, List[SafeTxID]] = {}
+
+    @property
+    def safe_details(self) -> SafeDetails:
+        slot = keccak(text="fallback_manager.handler.address")
+        value = self.provider.get_storage_at(self.contract.address, slot)
+        fallback_address = self.network_manager.ecosystem.decode_address(value[-20:])
+
+        return SafeDetails(
+            address=self.contract.address,
+            nonce=self.get_next_nonce(),
+            threshold=self.contract.getThreshold(),
+            owners=self.contract.getOwners(),
+            masterCopy=self.contract.masterCopy(),
+            modules=self.contract.getModules() if hasattr(self.contract, "getModules") else [],
+            # TODO: Add fallback handler getter
+            fallbackHandler=fallback_address,
+            guard=self.contract.getGuard() if hasattr(self.contract, "getGuard") else ZERO_ADDRESS,
```

Comment:
**suggestion**: Check for `getGuard` in the `self.contract._view_methods_` instead of `getattr`, it is a nicer API and more performant and readable.

See `get_next_nonce` is an example.

```python
if "getGuard" in self.contract._view_methods_`
```